homeassistant/components/switcher_kis/__init__.py

```diff
@@ -45,14 +45,19 @@ def on_device_data_callback(device: SwitcherBase) -> None:
 
         # New device - create device
         _LOGGER.info(
-            "Discovered Switcher device - id: %s, key: %s, name: %s, type: %s (%s)",
+            "Discovered Switcher device - id: %s, key: %s, name: %s, type: %s (%s), is_token_needed: %s",
             device.device_id,
             device.device_key,
             device.name,
             device.device_type.value,
             device.device_type.hex_rep,
+            device.token_needed,
         )
 
+        if device.token_needed and not entry.data.get(CONF_TOKEN):
```

Comment:
Since this can happen asynchronously during startup please store the constant locally (at line ~30):
`token = entry.data.get(CONF_TOKEN)`

```suggestion
        if device.token_needed and not token:
```

Reply:
It is done.